[CI] MlDistributedFailureIT testFullClusterRestart with IndexingPressure is adjusted twice

**Build scan**: https://gradle-enterprise.elastic.co/s/f7pp3lrzj2qko

**Repro line**: `./gradlew ':x-pack:plugin:ml:internalClusterTest' --tests "org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.testFullClusterRestart" -Dtests.seed=83C9FE9E2AF17826 -Dtests.security.manager=true -Dtests.locale=ga-IE -Dtests.timezone=Europe/Rome -Druntime.java=11`

**Reproduces locally?**: No

**Applicable branches**: only master

**Failure history**: Failed a total of 4 times, **since yesterday**
 https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=Europe/Bucharest&tests.container=org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT&tests.sortField=FAILED&tests.test=testFullClusterRestart&tests.unstableOnly=true
**Failure excerpt**:
```
        Caused by:
        java.lang.AssertionError: IndexingPressure is adjusted twice
            at __randomizedtesting.SeedInfo.seed([83C9FE9E2AF17826]:0)
            at org.elasticsearch.index.IndexingPressure.lambda$wrapReleasable$0(IndexingPressure.java:78)
            at org.elasticsearch.action.ActionListener$5.onFailure(ActionListener.java:303)
            at org.elasticsearch.action.support.replication.TransportReplicationAction$AsyncPrimaryAction.onFailure(TransportReplicationAction.java:445)
            at org.elasticsearch.action.support.replication.TransportReplicationAction$AsyncPrimaryAction.handleException(TransportReplicationAction.java:439)
            at org.elasticsearch.action.support.replication.TransportReplicationAction$AsyncPrimaryAction.lambda$runWithPrimaryShardReference$3(TransportReplicationAction.java:424)
            at org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:136)
            at org.elasticsearch.action.ActionListener$MappedActionListener.onFailure(ActionListener.java:97)
            at org.elasticsearch.action.support.replication.ReplicationOperation.finishAsFailed(ReplicationOperation.java:353)
            at org.elasticsearch.action.support.replication.ReplicationOperation.onNoLongerPrimary(ReplicationOperation.java:298)
            at org.elasticsearch.action.support.replication.ReplicationOperation$2.lambda$onFailure$2(ReplicationOperation.java:229)
            at org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:136)
            at org.elasticsearch.action.ActionListener.onFailure(ActionListener.java:247)
            at org.elasticsearch.action.ResultDeduplicator$CompositeListener.onFailure(ResultDeduplicator.java:116)
            at org.elasticsearch.cluster.action.shard.ShardStateAction$1.handleException(ShardStateAction.java:140)
            at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1186)
            at org.elasticsearch.transport.TransportService$7.doRun(TransportService.java:784)
            at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
            at org.elasticsearch.common.util.concurrent.EsExecutors$DirectExecutorService.execute(EsExecutors.java:176)
            at org.elasticsearch.transport.TransportService.sendRequestInternal(TransportService.java:764)
            at org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:663)
            at org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:589)
            at org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:574)
            at org.elasticsearch.cluster.action.shard.ShardStateAction.sendShardAction(ShardStateAction.java:126)
            at org.elasticsearch.cluster.action.shard.ShardStateAction.lambda$remoteShardFailed$0(ShardStateAction.java:177)
            at org.elasticsearch.action.ResultDeduplicator.executeOnce(ResultDeduplicator.java:51)
            at org.elasticsearch.cluster.action.shard.ShardStateAction.remoteShardFailed(ShardStateAction.java:175)
            at org.elasticsearch.action.support.replication.TransportWriteAction$WriteActionReplicasProxy.failShardIfNeeded(TransportWriteAction.java:450)
            at org.elasticsearch.action.support.replication.ReplicationOperation$2.onFailure(ReplicationOperation.java:228)
            at org.elasticsearch.action.support.RetryableAction$RetryingListener.onFinalFailure(RetryableAction.java:182)
            at org.elasticsearch.action.support.RetryableAction$RetryingListener.onFailure(RetryableAction.java:174)
            at org.elasticsearch.action.support.replication.TransportReplicationAction$ReplicasProxy.performOn(TransportReplicationAction.java:1080)
            at org.elasticsearch.action.support.replication.ReplicationOperation$3.tryAction(ReplicationOperation.java:244)
            at org.elasticsearch.action.support.RetryableAction$1.doRun(RetryableAction.java:99)
            at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
            at org.elasticsearch.common.util.concurrent.EsExecutors$DirectExecutorService.execute(EsExecutors.java:176)
            at org.elasticsearch.action.support.RetryableAction.run(RetryableAction.java:77)
            at org.elasticsearch.action.support.replication.ReplicationOperation.performOnReplica(ReplicationOperation.java:263)
            at org.elasticsearch.action.support.replication.ReplicationOperation.performOnReplicas(ReplicationOperation.java:192)
            at org.elasticsearch.action.support.replication.ReplicationOperation.handlePrimaryResult(ReplicationOperation.java:148)
            at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:128)
            at org.elasticsearch.action.ActionListener$MappedActionListener.onResponse(ActionListener.java:87)
            at org.elasticsearch.action.ActionListener.completeWith(ActionListener.java:351)
            at org.elasticsearch.action.bulk.TransportShardBulkAction$2.finishRequest(TransportShardBulkAction.java:225)
            at org.elasticsearch.action.bulk.TransportShardBulkAction$2.doRun(TransportShardBulkAction.java:194)
            at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
            at org.elasticsearch.action.bulk.TransportShardBulkAction.performOnPrimary(TransportShardBulkAction.java:230)
            at org.elasticsearch.action.bulk.TransportShardBulkAction.dispatchedShardOperationOnPrimary(TransportShardBulkAction.java:128)
            at org.elasticsearch.action.bulk.TransportShardBulkAction.dispatchedShardOperationOnPrimary(TransportShardBulkAction.java:86)
            at org.elasticsearch.action.support.replication.TransportWriteAction$1.doRun(TransportWriteAction.java:188)
            at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:739)
            at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
            at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
            at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
            at java.base/java.lang.Thread.run(Thread.java:834)
```

I have a feeling this got nothing to do with ML, but then I don't know how to label it otherwise.

--- a/server/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java
+++ b/server/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java
@@ -119,7 +119,7 @@ public class ReplicationOperation<
 
         totalShards.incrementAndGet();
         pendingActions.incrementAndGet(); // increase by 1 until we finish all primary coordination
-        primary.perform(request, ActionListener.wrap(this::handlePrimaryResult, resultListener::onFailure));
+        primary.perform(request, ActionListener.wrap(this::handlePrimaryResult, this::finishAsFailed));
     }
 
     private void handlePrimaryResult(final PrimaryResultT primaryResult) {

